lib/Sema/TypeCheckPattern.cpp

```diff
@@ -1590,12 +1590,13 @@ bool TypeChecker::coerceParameterListToType(ParameterList *P, ClosureExpr *CE,
   // the closure argument is also one to avoid the tuple splat
   // from happening.
   if (!hadError && isa<ParenType>(paramListType.getPointer())) {
-    auto underlyingTy = paramListType->getCanonicalType();
+    auto underlyingTy = cast<ParenType>(paramListType.getPointer())
+      ->getUnderlyingType();
     
-    if (underlyingTy->is<TupleType>()) {
-      if (P->size() == 1) {
+    if (underlyingTy->is<TupleType>() &&
+        !underlyingTy->castTo<TupleType>()->getVarArgsBaseType()) {
```

Comment:
I'm going to have to revisit some of this stuff anyway to fix the general case where we have multiple anonymous parameters, and one of them is variadic. My hope is that when we nail down SE-0110 better this logic can be greatly simplified.